lib/apiservers/engine/backends/volume.go

```diff
@@ -58,7 +105,43 @@ func (v *Volume) VolumeRm(name string) error {
 		if _, ok := err.(*storage.RemoveVolumeConflict); ok {
 			return derr.NewRequestConflictError(fmt.Errorf("Volume is in use"))
 		}
-		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error form portlayer"), http.StatusInternalServerError)
+		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error from portlayer: %s", err), http.StatusInternalServerError)
 	}
 	return nil
 }
+
+//Utility Functions
+
+func fillDockerVolumeModel(volume *models.VolumeResponse, labels map[string]string) *types.Volume {
+	dockerVol := types.Volume{
+		Driver:     volume.Driver,
+		Name:       volume.Name,
+		Labels:     labels,
+		Mountpoint: volume.SelfLink,
+	}
+	return &dockerVol
+}
+
+func translateInputsToPortlayerRequestModel(name, host, driverName string, opts, labels map[string]string) models.VolumeRequest {
+	model := models.VolumeRequest{
+		Driver:     driverName,
+		DriverArgs: opts,
+		Name:       name,
+		Store:      host,
```

Comment:
Store is the name of the datastore which we want to create a volume on.  Not the hostname or the uuid of the vch.

Think of it this way-  I have 3 datastores in my vsphere environment.  One of them is called `Fast`, the other `Backedup`, and the last `Junk`.

On `Fast`, I (as a VI admin) want my volumes created in `[FastDatastore] /path/to/some/place/relevant/that/i/decided`.
On `Backedup`, I want my volumes created on `[BackedupDatastore] /some/other/location`.
On `Junk`, I want my volumes created on `[JunkDatastore] /junk/folder`.

Anything *beyond* those paths are irrelevant to me.

How does a docker user now specify that stuff?  Im now a docker user and I just want to create a volume and know nothing of paths or datastores or whatever. 

We want to use `volumeStore` (or `Store` in this case) to encompass the datastore + path so the docker user doesnt need to know about the details.  

The VI admin would do the following when setting up the vch

`vic-admin create <....> -volumestore Fast="[FastDatastore] /path/to/where/i/want/volumes`

And the docker user would do the following.
`docker volume create --driver=vsphere --name="nameRelevantToMeTheDockerUser" -o "volumestore=Fast"`

Make sense?